kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java

```diff
@@ -108,13 +110,24 @@ public void start(Map<String, String> props) {
           url
       );
       if (!storage.bucketExists()) {
-        throw new DataException("No-existent S3 bucket: " + connectorConfig.getBucketName());
+        throw new ConnectException("Non-existent S3 bucket: " + connectorConfig.getBucketName());
       }
 
       writerProvider = newRecordWriterProvider(connectorConfig);
       partitioner = newPartitioner(connectorConfig);
 
       open(context.assignment());
+      try {
+        reporter = context.errantRecordReporter();
+        if (reporter == null) {
+          log.info("Errant record reporter not configured.");
+        }
+      } catch (NoSuchMethodError | NoClassDefFoundError | UnsupportedOperationException e) {
```

Comment:
```suggestion
      } catch (NoSuchMethodError | NoClassDefFoundError e) {
```

we don't need this IIUC

Reply:
You're right that for the reporter the `UnsupportedOperationException`  is not needed. This is needed in storage connectors that use the `StorageSinkTestBase` class in tests, thrown [here](https://github.com/confluentinc/kafka-connect-storage-common/blob/e1aa07bdbcd031d8f26a577a7bf3984b4a9ca243/core/src/main/java/io/confluent/connect/storage/StorageSinkTestBase.java#L254). The class is used in `S3SinkTaskTest`.